Groups

AlexChalk opened the pull request:
Users can create and join groups

dvidgry committed:
groups new created

BenNoonan1991 committed:
Adds tests to controller spec

BenNoonan1991 committed:
Adds feature test for creating a group

Commit:
Site user can create a group

Commit:
Add feature to test user can join groups

BenNoonan1991 committed:
Added group name button to group page

Commit:
Join Group: Beging adding functionality

BenNoonan1991 committed:
Altered show method in group controller

Commit:
Groups: User can join a group

Commit:
Merge remote-tracking branch 'origin/master' into create_groups

Commit:
Remove redundant tests

Commit:
Merge remote-tracking branch 'origin' into create_groups

enonnai commented:
Guys, I created a group with useraccount1 and joined, then I logged in with useraccount2 and clicked on the group name(step1 screenshot), and the following error occurs as it points to a wrong path groups/groups/1 (step2 screenshot):
![step_1](https://cloud.githubusercontent.com/assets/24532918/26723200/493b19e2-478b-11e7-8e0b-7f187245a035.png)
![step_2](https://cloud.githubusercontent.com/assets/24532918/26723209/4e2395a6-478b-11e7-9d34-983122eab4e2.png)

AlexChalk commented:
Error occurs when user navigates to `groups/` instead of `groups`.

Commit:
Remove navigate_to groups/groups/ bug

Bug occurred when there was a trailing slash in the address bar

AlexChalk commented:
@enonnai: Bug has been fixed!

Commit:
Update feature test to protect against slash bug

enonnai commented:
Awesome inter-teams work guys!

DSeanGray commented:
hey, guys! code looks good there are a lot of empty or unnecessary files and commented out code in here though. can you remove them please. then it's all good to go 👍

dvidgry reviewed (dismissed):
please remove the commented out code and redundant files and it all god to go :) good work guys 👍

AlexChalk commented:
@DSeanGray the stuff you're referring to is auto-generated so i'd rather leave it in.

anaalta reviewed (approved):
Looks good (with David's remark that there are some files with commented code only - the css, the js files)
:shipit:

BenNoonan1991 committed:
Merge remote-tracking branch 'origin' into create_groups

Commit:
Merge branch 'master' into create_groups

enonnai reviewed (approved):
Good stuff :shipit:

Commit:
Merge branch 'master' into create_groups

dbugsy commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,32 @@
+    @users = @group.users
+  end
+
+  def update
```
This is not updating a Group, it is creating a Membership. Also, it is non-idempotent, meaning you could call this action twice and you would get 2 similar Membership rows. PUT http requests should be idempotent. http://restcookbook.com/HTTP%20Methods/put-vs-post/

Not a blocker, but feels un-railsy - you could consider a refactor.

BenNoonan1991 committed:
Adds feature test for groups link

dbugsy commented on app/models/membership.rb:
```diff
@@ -0,0 +1,5 @@
+class Membership < ApplicationRecord
+  belongs_to :user
+  belongs_to :group
+  validates_uniqueness_of :user_id, scope: :group_id
```
very railsy - great!

dbugsy commented on spec/web_helpers.rb:
```diff
@@ -23,6 +23,13 @@ def create_a_post(text)
   click_button "Submit"
 end
 
+def create_a_group(title)
```
nice use of argument

dbugsy reviewed (approved):
Happy to let this through, but you could definitely improve this as follows:

`#update` Group is a little strange - looks like you could use the Membership `#create` action instead?

Commented code is a problem, there are a bunch of generated files here that are unused. It's good practice to delete those.

Commit:
Merge branch 'master' into create_groups

Commit:
Merge remote-tracking branch 'origin' into create_groups

Commit:
Merge remote-tracking branch 'origin' into create_groups

Commit:
Refactor group joining process

-membership handled via memberships#create instead of
 groups#update
-numerous empty files deleted

Commit:
Delete more redundant files

Commit:
Merge branch 'create_groups' of github.com:makersacademy/acebook-remote-april-2017 into create_groups
